src/Homepage/Button/SearchButton.jsx

```diff
@@ -0,0 +1,29 @@
+import React, { Component } from 'react';
+import './_search-button.scss';
+
+import { Link } from 'react-router-dom';
+
+
+
+export default class SearchButton extends Component {
+
+    shouldComponentUpdate(nextProps, nextState, nextContext) {
+        return this.props.query !== nextProps.query || this.props.selectedOption !== nextProps.selectedOption;
+    }
+
+    render() {
+
+        let url = `/image-list/${this.props.query}`;
+        if (this.props.selectedOption !== null) {
```

Comment:
After the destruction, the following line would do the same thing;

```suggestion
        if (selectedOption) {
```

Let's replace with this.